Source/Blazorise/Base/BaseInputComponent.razor.cs

```diff
@@ -89,21 +89,37 @@ protected override void Dispose( bool disposing )
         {
             if ( disposing )
             {
-                if ( ParentValidation != null )
-                {
-                    // To avoid leaking memory, it's important to detach any event handlers in Dispose()
-                    ParentValidation.ValidationStatusChanged -= OnValidationStatusChanged;
-                }
+                ReleaseResources();
+            }
 
-                ParentModal?.NotifyFocusableComponentRemoved( this );
+            base.Dispose( disposing );
+        }
 
-                if ( Theme != null )
-                {
-                    Theme.Changed -= OnThemeChanged;
-                }
+        /// <inheritdoc/>
+        protected override ValueTask DisposeAsync( bool disposing )
+        {
+            if ( disposing )
+            {
+                ReleaseResources();
             }
 
-            base.Dispose( disposing );
+            return base.DisposeAsync( disposing );
+        }
+
+        protected void ReleaseResources()
```

Comment:
The `BaseInputComponent` is marked only with `IDisposable`. So I don't think we need to overload `DisposeAsync`.

Reply:
Hmmm, actually I understand what you wanted to. If any of the sub-components has an `IAsyncDisposable` then it would pass through both methods. Yeah, makes sense.

Reply:
Exactly... 🤭